lib/ts/recipe/passwordless/api/implementation.ts

```diff
@@ -2,156 +2,810 @@ import { APIInterface } from "../";
-                input.options.req,
-                input.options.res,
-                input.tenantId,
-                loginMethod.recipeUserId,
-                {},
-                {},
-                input.userContext
-            );
+                if (linkRes.status !== "OK") {
+                    throw new RecurseError();
+                }
 
-            return {
-                status: "OK",
-                createdNewRecipeUser: response.createdNewRecipeUser,
-                user: response.user,
-                session,
+                let user = await getUser(recipeUserId.getAsString(), userContext);
+                if (user === undefined) {
+                    // linked user not found
+                    throw new SessionError({
+                        type: SessionError.UNAUTHORISED,
+                        message: "User not found",
+                    });
+                }
+
+                return user;
             };
+
+            const assertThatFactorUserBeingCreatedCanBeLinkedWithSessionUser = async (
+                tenantId: string,
+                sessionUser: User,
+                accountInfo: { email?: string; phoneNumber?: string },
+                userContext: UserContext
+            ) => {
+                if (!sessionUser.isPrimaryUser) {
+                    const canCreatePrimary = await AccountLinking.getInstance().recipeInterfaceImpl.canCreatePrimaryUser(
+                        {
+                            recipeUserId: sessionUser.loginMethods[0].recipeUserId,
+                            userContext,
+                        }
+                    );
+
+                    if (canCreatePrimary.status === "RECIPE_USER_ID_ALREADY_LINKED_WITH_PRIMARY_USER_ID_ERROR") {
+                        // Session user is linked to another primary user, which means the session is revoked as well
+                        throw new SessionError({
+                            type: SessionError.UNAUTHORISED,
+                            message: "Session may be revoked",
+                        });
+                    }
+
+                    if (
+                        canCreatePrimary.status === "ACCOUNT_INFO_ALREADY_ASSOCIATED_WITH_ANOTHER_PRIMARY_USER_ID_ERROR"
+                    ) {
+                        throw new SignInUpError({
+                            status: "SIGN_IN_UP_NOT_ALLOWED",
+                            reason:
+                                "Cannot complete MFA because of security reasons. Please contact support. (ERR_CODE_014)",
+                        });
+                    }
+                }
+
+                // Check if the linking with session user going to fail and avoid user creation here
+                const users = await AccountLinking.getInstance().recipeInterfaceImpl.listUsersByAccountInfo({
+                    tenantId,
+                    accountInfo,
+                    doUnionOfAccountInfo: true,
```

Comment:
why true here, but false in emailpassword recipe?

Reply:
Does not matter, have made it false to be consistent